src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs

```diff
@@ -50,6 +50,8 @@ internal abstract class ServiceConnectionContainerBase : IServiceConnectionConta
 
         private volatile ServiceConnectionStatus _status;
 
+        private bool _isOffline = false;
```

Comment:
Shall we use single flag to make this status check? Like merge _isOffline and _terminated? Looks both terminate all connections and hold on for new ones.

Reply:
They indeed serve the same purpose but at different stages of shutdown. Perhaps the only reason leave them separate is for debugging purpose? Having one flag is cleaner though.